native/math32.go

```diff
@@ -0,0 +1,138 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// float32 version of standard library math package routines used here.
+
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"math"
+)
+
+const (
+	uvnan    = 0x7fc00000
```

Comment:
At some stage math32 may want to move into gonum/internal, but we can think about that if other paskages start to use float32 routines.